azure/scope/managedcontrolplane.go

```diff
@@ -328,7 +351,17 @@ func (s *ManagedControlPlaneScope) IsIPv6Enabled() bool {
 
 // IsVnetManaged returns true if the vnet is managed.
 func (s *ManagedControlPlaneScope) IsVnetManaged() bool {
-	return true
+	if s.cache != nil && s.cache.isVnetManaged != nil {
+		return to.Bool(s.cache.isVnetManaged)
+	}
+	ctx := context.Background()
+	ctx, log, done := tele.StartSpanWithLogger(ctx, "scope.ManagedControlPlaneScope.IsVnetManaged")
+	defer done()
+	isManaged, err := virtualnetworks.New(s).IsManaged(ctx)
+	if err != nil {
+		log.Error(err, "Unable to determine if ManagedControlPlaneScope VNET is managed by capz", "AzureManagedCluster", s.ClusterName())
+	}
+	return isManaged
```

Comment:
can we also set the cache here if it's unset?

Reply:
(only if you go with approach b) from https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/2543/files#r946138219)

Reply:
+1 to being cached instead of making frequent API calls during the reconcile loop.

Reply:
done